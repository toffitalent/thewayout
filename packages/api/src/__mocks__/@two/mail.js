const send = jest.fn().mockResolvedValue(undefined);

module.exports = {
  send,
};
