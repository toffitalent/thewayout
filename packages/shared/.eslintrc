{
  "settings": {
    "import/resolver": {
      "typescript": {
        "project": ["tsconfig.json", "packages/shared/tsconfig.json"]
      }
    }
  },
  "rules": {
    "no-param-reassign": "off",
    "react/react-in-jsx-scope": "off"
  }
}
