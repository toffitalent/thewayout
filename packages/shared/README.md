# @two/shared

Copyright (c) 2023-Present The Way Out, Inc. All Rights Reserved.

## Development

This package contains code shared between the [api](../api/README.md), [app](../app/README.md), and [web](../web/README.md) packages and is not intended to run in a standalone manner. See the READMEs of those packages for information on starting a development environment.

## Testing

To run all component and unit tests and get coverage reports, simply run `yarn workspace @dlatwobs/shared test`.
