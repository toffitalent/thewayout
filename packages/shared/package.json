{
  "name": "@two/shared",
  "version": "1.0.0",
  "author": "Disruptive Labs",
  "license": "UNLICENSED",
  "private": true,
  "main": "build/index.js",
  "types": "build/index.d.ts",
  "scripts": {
    "build": "tsc -b tsconfig.build.json",
    "clean": "rimraf build coverage",
    "test": "jest --coverage",
    "type-check": "tsc"
  },
  "dependencies": {
    "@disruptive-labs/ui": "file:../../vendor/@disruptive-labs/ui",
    "axios": "^0.27.2",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@testing-library/dom": "^9.3.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/jest": "^29.5.2",
    "@types/react": "^18.2.14",
    "babel-jest": "^29.5.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.5.0",
    "jest-environment-jsdom": "^29.5.0",
    "jest-junit": "^16.0.0",
    "react-test-renderer": "^18.2.0",
    "rimraf": "^3.0.2",
    "sass": "^1.63.6",
    "typescript": "^5.2.2"
  },
  "jest": {
    "clearMocks": true,
    "collectCoverageFrom": [
      "<rootDir>/src/**/*.{ts,tsx}"
    ],
    "globalSetup": "<rootDir>/test/setup.js",
    "moduleNameMapper": {
      "^.+\\.module\\.(css|sass|scss)$": "identity-obj-proxy",
      "^.+\\.(css|sass|scss)$": "<rootDir>/test/styleMock.js",
      "^.+\\.(jpg|jpeg|png|gif|webp|svg)$": "<rootDir>/test/fileMock.js",
      "^@test$": "<rootDir>/test/index.tsx"
    },
    "setupFilesAfterEnv": [
      "<rootDir>/test/jest.test.ts"
    ],
    "testEnvironment": "jsdom",
    "testPathIgnorePatterns": [
      "<rootDir>/node_modules/",
      "<rootDir>/test/"
    ],
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": [
        "babel-jest",
        {
          "presets": [
            "next/babel"
          ]
        }
      ]
    },
    "transformIgnorePatterns": [
      "node_modules/(?!(@?disruptive-labs.*|@popperjs/core|es-cookie)/)"
    ]
  },
  "jest-junit": {
    "outputDirectory": "./artifacts/jest",
    "outputName": "results.xml"
  }
}
