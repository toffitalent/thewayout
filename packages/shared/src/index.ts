// NOTE: React components should not exported here

export * from './constants';
export * from './types';
export * from './utils';
