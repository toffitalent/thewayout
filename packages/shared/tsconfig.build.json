{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "composite": true,
    "incremental": true,
    "module": "commonjs",
    "noEmit": false,
    "outDir": "./build",
    "rootDir": "./src",
    "tsBuildInfoFile": "./build/.tsbuildinfo"
  },
  "include": ["**/*.ts", "**/*.tsx"],
  "exclude": ["build", "node_modules", "**/*.test.ts", "**/*.test.tsx", "test"]
}
