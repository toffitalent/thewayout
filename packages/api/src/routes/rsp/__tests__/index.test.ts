import { faker } from '@faker-js/faker';
import { JusticeStatus, OffenseCategory, Support, VeteranOrJustice } from '@two/shared';
import { auth, fixtures, request, resetDb } from '@test';

describe('Routes > Rsp', () => {
  beforeEach(async () => {
    await resetDb();

    faker.seed(1);
  });

  describe('POST /api/v1/rsp', () => {
    const data = {
      name: 'Organization Name',
      description: '',
      address: '',
      city: '',
      state: '',
      postalCode: '',
      phone: '',
      email: '',
      servicesArea: [],
      support: [Support.basicHygieneSupport],
      veteranOrJustice: [VeteranOrJustice.justiceImpacted],
      justiceStatus: [JusticeStatus.extendedSupervision],
      offenses: [OffenseCategory.arson],
      owner: {
        phone: '',
        position: '',
      },
    };

    test('creates rsp organization and updates owner account', async () => {
      expect(fixtures.rspAccount3.isProfileFilled).toBeFalsy();

      const invalidRes = await request()
        .post('/api/v1/rsp')
        .use(auth(fixtures.user9.id, ['rsp']))
        .send({ ...data, justiceStatus: undefined, offenses: undefined });
      expect(invalidRes.status).toBe(400);

      const res = await request()
        .post('/api/v1/rsp')
        .use(auth(fixtures.user9.id, ['rsp']))
        .send(data);

      expect(res.status).toBe(200);
      expect(res.body.owner.isProfileFilled).toBeTruthy();
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot({
        rsp: {
          id: expect.any(String),
        },
        owner: {
          id: expect.any(String),
          rspId: expect.any(String),
          rsp: {
            id: expect.any(String),
          },
        },
      });
    });

    test('creates rsp organization only with unique name', async () => {
      const res = await request()
        .post('/api/v1/rsp')
        .use(auth(fixtures.user7.id, ['rsp']))
        .send({ ...data, name: fixtures.rsp1.name });

      expect(res.status).toBe(409);
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot();
    });

    test('creates rsp organization only by owner account', async () => {
      const res = await request()
        .post('/api/v1/rsp')
        .use(auth(fixtures.user8.id, ['rsp']))
        .send(data);

      expect(res.status).toBe(404);
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot();
    });

    test('creates rsp organization - allowed veteran client type and updates owner account', async () => {
      expect(fixtures.rspAccount3.isProfileFilled).toBeFalsy();

      const res = await request()
        .post('/api/v1/rsp')
        .use(auth(fixtures.user9.id, ['rsp']))
        .send({
          ...data,
          veteranOrJustice: [VeteranOrJustice.veteran],
          justiceStatus: undefined,
          offenses: undefined,
        });

      expect(res.status).toBe(200);
      expect(res.body.owner.isProfileFilled).toBeTruthy();
      expect(res.type).toBe('application/json');
      expect(res.body).toMatchSnapshot({
        rsp: {
          id: expect.any(String),
        },
        owner: {
          id: expect.any(String),
          rspId: expect.any(String),
          rsp: {
            id: expect.any(String),
          },
        },
      });
    });
  });
});
