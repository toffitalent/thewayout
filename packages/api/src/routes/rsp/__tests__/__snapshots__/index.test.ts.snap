// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes > Rsp POST /api/v1/rsp creates rsp organization - allowed veteran client type and updates owner account 1`] = `
{
  "owner": {
    "avatar": null,
    "id": Any<String>,
    "isProfileFilled": true,
    "phone": "",
    "role": "owner",
    "rsp": {
      "address": "",
      "avatar": null,
      "city": "",
      "createdAt": "<Date>",
      "description": "",
      "email": "",
      "id": Any<String>,
      "name": "Organization Name",
      "phone": "",
      "postalCode": "",
      "servicesArea": [],
      "state": "",
      "support": [
        "basicHygieneSupport",
      ],
      "updatedAt": "<Date>",
      "veteranOrJustice": [
        "veteran",
      ],
    },
    "rspId": Any<String>,
    "userId": "01817e24-0d42-761b-8f08-5f532d6045f9",
  },
  "rsp": {
    "address": "",
    "avatar": null,
    "city": "",
    "createdAt": "<Date>",
    "description": "",
    "email": "",
    "id": Any<String>,
    "name": "Organization Name",
    "phone": "",
    "postalCode": "",
    "servicesArea": [],
    "state": "",
    "support": [
      "basicHygieneSupport",
    ],
    "updatedAt": "<Date>",
    "veteranOrJustice": [
      "veteran",
    ],
  },
}
`;

exports[`Routes > Rsp POST /api/v1/rsp creates rsp organization and updates owner account 1`] = `
{
  "owner": {
    "avatar": null,
    "id": Any<String>,
    "isProfileFilled": true,
    "phone": "",
    "role": "owner",
    "rsp": {
      "address": "",
      "avatar": null,
      "city": "",
      "createdAt": "<Date>",
      "description": "",
      "email": "",
      "id": Any<String>,
      "justiceStatus": [
        "extendedSupervision",
      ],
      "name": "Organization Name",
      "offenses": [
        "arson",
      ],
      "phone": "",
      "postalCode": "",
      "servicesArea": [],
      "state": "",
      "support": [
        "basicHygieneSupport",
      ],
      "updatedAt": "<Date>",
      "veteranOrJustice": [
        "justiceImpacted",
      ],
    },
    "rspId": Any<String>,
    "userId": "01817e24-0d42-761b-8f08-5f532d6045f9",
  },
  "rsp": {
    "address": "",
    "avatar": null,
    "city": "",
    "createdAt": "<Date>",
    "description": "",
    "email": "",
    "id": Any<String>,
    "justiceStatus": [
      "extendedSupervision",
    ],
    "name": "Organization Name",
    "offenses": [
      "arson",
    ],
    "phone": "",
    "postalCode": "",
    "servicesArea": [],
    "state": "",
    "support": [
      "basicHygieneSupport",
    ],
    "updatedAt": "<Date>",
    "veteranOrJustice": [
      "justiceImpacted",
    ],
  },
}
`;

exports[`Routes > Rsp POST /api/v1/rsp creates rsp organization only by owner account 1`] = `
{
  "code": "ResourceNotFound",
  "message": "Rsp user not found",
  "status": 404,
}
`;

exports[`Routes > Rsp POST /api/v1/rsp creates rsp organization only with unique name 1`] = `
{
  "code": "ResourceConflict",
  "message": "Rsp name already exist",
  "status": 409,
}
`;
